src/hooks/hookProxy.ts

```diff
@@ -1,79 +1,496 @@
+                  remsUrl = referenceValue;
+                } else if (referenceValue.$ && referenceValue.$.value) {
+                  // Object with value attribute
+                  remsUrl = referenceValue.$.value;
+                }
+                
+                if (remsUrl) {
+                  // Extract the URL, handling different formats
+                  let cdsServiceUrl;
+                  if (remsUrl.includes(':rems_discovery:')) {
+                    cdsServiceUrl = remsUrl.split(':rems_discovery:')[1];
+                  } else {
+                    cdsServiceUrl = remsUrl;
+                  }
+                  
+                  // Create the return object with drug info and endpoints
+                  const drugInfo = {
+                    brandName: brandName,
+                    genericName: genericName,
+                    remsEndpoint: cdsServiceUrl,                  };
+                  
+                  console.log(`Found REMS endpoint for ${brandName}: ${cdsServiceUrl}`);
+                  return drugInfo;
+                }
+              }
+            }
+          }
+          
+          // Found the drug but no REMS endpoint
+          console.log(`Found drug ${brandName} but no REMS endpoint in XML`);
+          return {
+            brandName: brandName,
+            genericName: genericName,
+            remsEndpoint: null,
+            etasuEndpoint: null
+          };
+        }
+      }
+    }
+    
+    console.log(`Drug ${searchDrugName} not found in XML`);
+    return null;
+  } catch (error) {
+    console.error('Error extracting drug info from XML:', error);
+    return null;
+  }
+}
+
+async function getDrugXmlFromSplZip(rems_spl_date?: string): Promise<any> {
+  // Create base directories to store and process the zip files
+  const baseFilePath = join(process.cwd(), 'rems-spl-files');
+  const extractPath = join(baseFilePath, 'extracted');
+  const innerExtractPath = join(baseFilePath, 'inner_extracted');
+  const mainZipPath = join(baseFilePath, REMSAdminWhitelist.zipFileName);
+  
+  // Create directories if they don't exist
+  if (!fs.existsSync(baseFilePath)) {
+    fs.mkdirSync(baseFilePath, { recursive: true });
+  }
+  if (!fs.existsSync(extractPath)) {
+    fs.mkdirSync(extractPath, { recursive: true });
+  }
+  if (!fs.existsSync(innerExtractPath)) {
+    fs.mkdirSync(innerExtractPath, { recursive: true });
+  }
+  
+  try {
+    // Check if we need to download the main zip file
+    let mainZipExists = fs.existsSync(mainZipPath);
+    let mainZipExtracted = fs.existsSync(join(extractPath, 'rems_document_spl_files')) && 
+                          fs.readdirSync(join(extractPath, 'rems_document_spl_files')).length > 0;
+    
+    // 1. Download and extract the main zip if needed
+    if (!mainZipExists || !mainZipExtracted) {
+      if (!mainZipExists) {
+        console.log(`Main SPL zip not found. Downloading from ${REMSAdminWhitelist.discoveryUrl}/drugs/spl.zip`);
```

Comment:
I think the /drugs/ path should be part of the discovery url. This would allow for a fully custom path to the file, not just at discoveryUrl/drugs/spl.zip.

Reply:
The full path to the zip should be configurable. Is the other zip filename because that is the name of the file downloaded or extracted? Ideally the filenames should be the same.

Reply:
this has been configured